shadows/framework/src/main/java/org/robolectric/shadows/util/SQLiteLibraryLoader.java

```diff
@@ -113,17 +119,34 @@ private String getOsPrefix() {
     } else if (name.contains("mac")) {
       return OS_MAC;
     } else {
-      throw new UnsupportedOperationException("Architecture '" + name + "' is not supported by SQLite library");
+      throw  new UnsupportedOperationException("Platform '" + name + "' is not supported by SQLite library");
     }
   }
 
-  private String getArchitectureSuffix() {
+  private String getArchitectureSuffix(String prefix) {
     String arch = System.getProperty("os.arch").toLowerCase(Locale.US).replaceAll("\\W", "");
-    if ("i386".equals(arch) || "x86".equals(arch)) {
-      return "x86";
-    } else {
-      return "x86_64";
+    switch (prefix) {
+      case OS_MAC:
+        return null;
+      case OS_LINUX:
+        switch (arch) {
+          case "i386":
+          case "x86":
+            return "i386";
+          case "x86_64":
+          case "amd64":
+            return "amd64";
+        }
```

Comment:
please add a break statement after this to avoid fallthrough (ErrorProne complains about this).